packages/core/src/semantic-icon-provider/SemanticIconProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  CalendarIcon,
+  ChevronDownIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronUpIcon,
+  CloseIcon,
+  ErrorSolidIcon,
+  InfoSolidIcon,
+  OverflowMenuIcon,
+  SuccessTickIcon,
+  TearOutIcon,
+  UserSolidIcon,
+  WarningSolidIcon,
+} from "@salt-ds/icons";
+import {
+  type ElementType,
+  type ReactNode,
+  createContext,
+  useContext,
+} from "react";
+
+export type SemanticIconMap = {
+  ChevronDownIcon: ElementType;
+  ChevronDownFilledIcon: ElementType;
+  ChevronUpIcon: ElementType;
+  ChevronUpFilledIcon: ElementType;
+  ErrorSolidIcon: ElementType;
+  SuccessTickIcon: ElementType;
+  InfoSolidIcon: ElementType;
+  WarningSolidIcon: ElementType;
+  ChevronRightIcon: ElementType;
+  OverflowMenuIcon: ElementType;
+  UserSolidIcon: ElementType;
+  CalendarIcon: ElementType;
+  ChevronLeftIcon: ElementType;
+  CloseIcon: ElementType;
+  ChevronRightFilledIcon: ElementType;
+  TearOutIcon: ElementType;
+};
+
+export interface SemanticIconProviderProps {
+  /**
+   * Custom mapping of icon names to components. Overrides default icons if provided.
+   */
+  iconMap?: Partial<SemanticIconMap>;
+
+  /**
+   * Child elements that will use the provided icons.
+   */
+  children: ReactNode;
+}
+
+const defaultIconMap: SemanticIconMap = {
```

Comment:
Some of the name is not matching the type?

Reply:
nope, we need both filled and unfilled in PEPPER, but in Salt those are unFilled only